### PR TITLE
Update to TerraForm v0.9.4

### PR DESCRIPTION
**Pull request by alexsomesan**
Converts ignition resources to data-sources.
This is a first step in preparation of adopting Terraform 0.9.

We CANNOT MOVE yet because of a known issue with ignition provider that wasn't resolved yet.
Tried to do a build with these changes. The first attempt fails with the error described in the issue below.
A second attempt builds successfully.

Issue tracker here: https://github.com/hashicorp/terraform/issues/11518

@Quentin-M @s-urbaniak

### CODE REVIEW

File: modules/azure/master/ignition-master.tf [hunk 1]
```diff
@@ -1,33 +1,41 @@
...
-    "${ignition_file.kubeconfig.id}",
-    "${ignition_file.kubelet-env.id}",
-    "${ignition_file.max-user-watches.id}",
+    "${data.ignition_file.kubeconfig.id}",
+    "${data.ignition_file.kubelet-env.id}",
+    "${data.ignition_file.max-user-watches.id}",
   ]
 
   systemd = [
+<<<<<<< HEAD
```
Comment (Quentin-M) [inline modules/azure/master/ignition-master.tf]: ?

File: modules/azure/master/ignition-master.tf [hunk 2]
```diff
@@ -1,33 +1,41 @@
...
     "${ignition_systemd_unit.kubelet-master.id}",
     "${ignition_systemd_unit.tectonic.id}",
     "${ignition_systemd_unit.bootkube.id}",
+=======
+    "${data.ignition_systemd_unit.etcd-member.id}",
+    "${data.ignition_systemd_unit.docker.id}",
+    "${data.ignition_systemd_unit.locksmithd.id}",
+    "${data.ignition_systemd_unit.kubelet-master.id}",
+    "${data.ignition_systemd_unit.tectonic.id}",
+>>>>>>> Convert ignition resources to datasources. Prepare for moving to TF 0.9
```
Comment (Quentin-M) [inline modules/azure/master/ignition-master.tf]: ?

**Comment by Quentin-M** [comment]
Hey!

@alexsomesan Is this PR still good? I want to re-use it alongside with a v0.9.4 bump.

**Comment by alexsomesan** [comment]
Sorry about that forgotten conflict :)
Should be good now. Also rebased against current master.

**Comment by Quentin-M** [comment]
Thanks for the update! 👍  👍  👍 
Let me update some other stuff in there!

**Comment by alexsomesan** [comment]
Have fun! ;)
going to bed now...

**Comment by Quentin-M** [comment]
Successfully deployed a cluster with a clean build of this PR and TerraForm v0.9.4. Good to merge!

**Comment by yifan-gu** [comment]
cc @derekparker for the builder image bump. (1.7 after this PR gets merged)

**Comment by Quentin-M** [comment]
Tests passed despite quota issue, also deployed clusters manually.

**Comment by Quentin-M** [comment]
Thanks!

**Pull request merged**